File: scheduler/core/src/main/java/io/orkes/conductor/dao/scheduler/NoOpSchedulerCacheDAO.java
package io.orkes.conductor.dao.scheduler;

import io.orkes.conductor.scheduler.model.WorkflowScheduleModel;

/**
 * No-op implementation of {@link SchedulerCacheDAO} that always reports a cache miss. Used as the
 * default when no external cache (e.g. Redis) is configured.
 */
public class NoOpSchedulerCacheDAO implements SchedulerCacheDAO {

    @Override
    public void updateSchedule(WorkflowScheduleModel workflowSchedule) {}

    @Override
    public WorkflowScheduleModel findScheduleByName(String name) {
        return null;
    }

    @Override
    public boolean exists(String name) {
        return false;
    }

    @Override
    public void deleteWorkflowSchedule(String name) {}

    @Override
    public long getNextRunTimeInEpoch(String scheduleName) {
        return -1L;
    }

    @Override
    public void setNextRunTimeInEpoch(String scheduleName, long epochMilli) {}
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/dao/scheduler/NoOpSchedulerCacheDAO.ts
import { SchedulerCacheDAO } from './SchedulerCacheDAO';
import { WorkflowScheduleModel } from '../../scheduler/model/WorkflowScheduleModel';

/**
 * No-op implementation of SchedulerCacheDAO that always reports a cache miss. Used as the
 * default when no external cache (e.g. Redis) is configured.
 */
export class NoOpSchedulerCacheDAO implements SchedulerCacheDAO {

    public updateSchedule(workflowSchedule: WorkflowScheduleModel): void {}

    public findScheduleByName(name: string): WorkflowScheduleModel | null {
        return null;
    }

    public exists(name: string): boolean {
        return false;
    }

    public deleteWorkflowSchedule(name: string): void {}

    public getNextRunTimeInEpoch(scheduleName: string): number {
        return -1;
    }

    public setNextRunTimeInEpoch(scheduleName: string, epochMilli: number): void {}
}
